src/tasks/withdraw.ts

```diff
@@ -187,17 +209,40 @@ async function getWithdrawals(
     if (balance.eq(0)) {
       continue;
     }
-    const usdValue = await oneinchUsdValue(token, balance, hre.network.name);
+    const pricedToken = await appraise(token, hre.network.name);
+    const balanceUsd = pricedToken.usdValue
+      .mul(balance)
+      .div(pricedToken.oneInWei);
     clearLine();
-    if (usdValue.gte(utils.parseUnits(minValue, DAI_DECIMALS))) {
-      withdrawals.push({ token, balance, usdValue });
-    } else {
+    if (
+      balanceUsd.lt(minValueWei.add(leftoverWei)) ||
+      (balanceUsd.isZero() && !(minValueWei.isZero() && leftoverWei.isZero()))
+    ) {
       console.warn(
         `Ignored ${utils.formatUnits(balance, token.decimals ?? 0)} units of ${
           token.symbol ?? "unknown token"
-        } (${token.address}) with value ${formatUsdValue(usdValue)} USD`,
+        } (${token.address}) with value ${formatUsdValue(balanceUsd)} USD`,
       );
+      continue;
+    }
+    let amount;
+    let amountUsd;
+    if (balanceUsd.isZero()) {
```

Comment:
Why would we withdraw if balanceUsd is zero? In case we don't have a price estimate?

Reply:
Yes, tokens with no price estimate have price zero, so the only way to withdraw them is in the case `minValueWei.isZero() && leftoverWei.isZero()`: I wanted to make sure that by not specifying any flag we would always withdraw every token in the contract. This to me makes sense because if there is no price restriction (`minValue` or `leftover`) then the price information (`...Usd`) should not matter. Not sure how to make it cleaner from the code, but I can sure add more comments.
Note that there are almost never** have withdrawals with zero amounts, since [all empty balances are skipped before](https://github.com/gnosis/gp-v2-contracts/pull/630/files#diff-d102d6e9ceccbd66e9c875a109481846a6d6f36ea30f1d0710ae161c3ab953a1R209-R211).

** I believe that with `minValue == 0` and `leftover` very close to `balance` the division might round down the withdrawn amount computation to zero, leading to a zero transfer. I'm ok with this since if you don't specify `minValue` then you probably don't care too much about gas prices.

Reply:
Maybe you can add this logic as a comment to the complicated if check (`minValueWei.isZero() && leftoverWei.isZero()`) above?